contracts/src/deployers/ezeth-linea/EzETHLineaHyperdriveDeployerCoordinator.sol

```diff
@@ -0,0 +1,179 @@
+
+    /// @notice The Renzo deposit contract on Linea. The latest mint rate is
+    ///         used as the vault share price.
+    IXRenzoDeposit public immutable xRenzoDeposit;
+
+    /// @notice Instantiates the deployer coordinator.
+    /// @param _name The deployer coordinator's name.
+    /// @param _factory The factory that this deployer will be registered with.
+    /// @param _coreDeployer The core deployer.
+    /// @param _target0Deployer The target0 deployer.
+    /// @param _target1Deployer The target1 deployer.
+    /// @param _target2Deployer The target2 deployer.
+    /// @param _target3Deployer The target3 deployer.
+    /// @param _target4Deployer The target4 deployer.
+    /// @param _xRenzoDeposit The xRenzoDeposit contract that provides the
+    ///        vault share price.
+    constructor(
+        string memory _name,
+        address _factory,
+        address _coreDeployer,
+        address _target0Deployer,
+        address _target1Deployer,
+        address _target2Deployer,
+        address _target3Deployer,
+        address _target4Deployer,
+        IXRenzoDeposit _xRenzoDeposit
+    )
+        HyperdriveDeployerCoordinator(
+            _name,
+            _factory,
+            _coreDeployer,
+            _target0Deployer,
+            _target1Deployer,
+            _target2Deployer,
+            _target3Deployer,
+            _target4Deployer
+        )
+    {
+        xRenzoDeposit = _xRenzoDeposit;
+    }
+
+    /// @dev Prepares the coordinator for initialization by drawing funds from
+    ///      the LP, if necessary.
+    /// @param _hyperdrive The Hyperdrive instance that is being initialized.
+    /// @param _lp The LP that is initializing the pool.
+    /// @param _contribution The amount of capital to supply. The units of this
+    ///        quantity are either base or vault shares, depending on the value
+    ///        of `_options.asBase`.
+    /// @param _options The options that configure how the initialization is
+    ///        settled.
+    /// @return value The value that should be sent in the initialize transaction.
+    function _prepareInitialize(
+        IHyperdrive _hyperdrive,
+        address _lp,
+        uint256 _contribution,
+        IHyperdrive.Options memory _options
+    ) internal override returns (uint256) {
+        // Depositing as base is disallowed.
+        if (_options.asBase) {
+            revert IHyperdrive.UnsupportedToken();
+        }
+
+        // Take custody of the contribution and approve Hyperdrive to pull
+        // the tokens.
+        IERC20 vaultSharesToken = IERC20(_hyperdrive.vaultSharesToken());
+        bool success = vaultSharesToken.transferFrom(
+            _lp,
+            address(this),
+            _contribution
+        );
+        if (!success) {
+            revert IHyperdriveDeployerCoordinator.TransferFailed();
+        }
+        success = vaultSharesToken.approve(address(_hyperdrive), _contribution);
```

Comment:
Why not safeTransferFrom and forceApprove?

Reply:
We talked about this offline. The gist is that the xezETH token is implemented using OpenZeppelin's token libraries. This means that we don't need the additional safety of `forceApprove` and `safeTransfer{From}`.